src/main/java/graphql/schema/GraphQLDirective.java

```diff
@@ -118,6 +118,37 @@ public Builder argument(GraphQLArgument fieldArgument) {
         }
 
         /**
+         * Take an argument builder in a function definition and apply. Can be used in a jdk8 lambda
+         * e.g.:
+         * <pre>
+         *     {@code
+         *      argument(a -> a.name("argumentName"))
+         *     }
+         * </pre>
+         *
+         * @param builderFunction a supplier for the builder impl
+         * @return this
+         */
+        public Builder argument(BuilderFunction<GraphQLArgument.Builder> builderFunction) {
+            GraphQLArgument.Builder builder = GraphQLArgument.newArgument();
+            builder = builderFunction.apply(builder);
+            this.arguments.add(builder.build());
+            return this;
+        }
```

Comment:
Could we call and return the `argument(GraphQLArgument.Builder builder)` overload that's defined below?